common/chat.ts

```diff
@@ -1,33 +1,80 @@
 import { SbUser, SbUserId } from './users/user-info'
 
-export enum ChatMessageType {
+/** Chat messages which are persisted in the DB and shown each time the user opens the app. */
+export enum ServerChatMessageType {
   TextMessage = 'message',
 }
 
-export interface BaseChatMessageData {
-  readonly type: ChatMessageType
+/** Chat messages which are only displayed on the client and are cleared when the app reloads. */
+export enum ClientChatMessageType {
+  JoinChannel = 'joinChannel',
+  LeaveChannel = 'leaveChannel',
+  NewChannelOwner = 'newOwner',
+  SelfJoinChannel = 'selfJoinChannel',
 }
 
-export interface ChatTextMessageData extends BaseChatMessageData {
-  type: typeof ChatMessageType.TextMessage
-  text: string
-}
+export type ChatMessageType = ServerChatMessageType | ClientChatMessageType
 
-export type ChatMessageData = ChatTextMessageData
+// TODO(2Pac): Include more information here, e.g. channel permissions, join date, etc.
+export interface ChatUser {
+  id: SbUserId
+  name: string
+}
 
-export interface ChatMessage {
+export interface BaseChatMessage {
   id: string
+  type: ChatMessageType
+  channel: string
+  time: number
+}
+
+/** A common text message that the user types in a channel. */
+export interface TextMessage extends BaseChatMessage {
+  type: typeof ServerChatMessageType.TextMessage
+  from: SbUserId
   user: ChatUser
-  sent: number
-  data: ChatMessageData
+  text: string
 }
 
-// TODO(2Pac): Include more information here, e.g. channel permissions, join date, etc.
-export interface ChatUser {
-  id: SbUserId
-  name: string
+/** A message that is displayed in the chat when someone joins the channel. */
+export interface JoinChannelMessage extends BaseChatMessage {
+  type: typeof ClientChatMessageType.JoinChannel
+  userId: SbUserId
```

Comment:
Unsure if these are new structures or moved from somewhere else or whatever, but probably this is not enough information for this particular message (this would be a user we do not necessarily have a name, etc. for, so it probably needs full info)

Reply:
Yeah, these are just the structures that were moved from the existing records that we used before. I didn't feel like reevaluating them at this moment to be honest, but you're right, this particular structure should include more information.

It's not so bad at the moment since we're returning user infos for all users when we retrieve the user list for a channel, but that doesn't cover the case of displaying this message when someone leaves the channel (what #758 enables), or even if we allow hiding the user list in some way in future.

Reply:
I actually think this is okay now that I'm looking at it today. I was thinking this was the event, but this is just the output of the event.

So the event itself will result in us population the user reducer state, this message just refers to that state indirectly (and referring to that indirectly is definitely better in case some state (e.g. their name) changes in the meantime)